src/Datagrid/SimplePager.php

```diff
@@ -77,6 +77,11 @@ public function getResults($hydrationMode = null)
         if (\count($this->results) > $this->getMaxPerPage()) {
             $this->haveToPaginate = true;
 
+            @trigger_error(
+                'The '.__METHOD__.' method will not return Doctrine\Common\Collections\ArrayCollection starting from 4.0',
```

Comment:
Shouldn't this deprecation be triggered even when `SimplePager::$results` is already set (https://github.com/sonata-project/SonataAdminBundle/pull/5138/files#diff-7b1b93d6330bfae84c4609b520fc1a6aR72)?
Instead of hardcoding the QCN, I'd use `ArrayCollection::class`.

Reply:
I don't think that's its a good idea to send the deprecation notice more then once for the same context.

Reply:
If the condition where this exception is placed is never reached, the user will not be notified about the deprecation.